App.js

```diff
@@ -1,86 +1,82 @@
-import {NavigationContainer, DefaultTheme, CommonActions, getStateFromPath, useFocusEffect} from '@react-navigation/native';
-import {Text, View, Image, Pressable, Linking, Platform, BackHandler, Dimensions} from 'react-native';
+import {NavigationContainer, DefaultTheme,  getStateFromPath, useFocusEffect,  createNavigationContainerRef, useNavigationState, stripKeysFromNavigationState } from '@react-navigation/native';
+import {Text, View, Image, Pressable,  BackHandler, Dimensions, Platform}  from 'react-native';
 import {createNativeStackNavigator} from '@react-navigation/native-stack';
 import Constants from 'expo-constants';
 import createWebNavigator from './createWebNavigator';
+import * as Linking from 'expo-linking';
 
 const chevronStyle = {width: 30, height: 30, resizeMode: 'contain', marginRight: 10};
 const chatTitleStyle = {fontSize: 18, fontWeight: 'bold'};
 const titleStyle = {fontSize: 24, fontWeight: 'bold', flex: 1};
 
 const config = {
-  initialRouteName: '',
+  initialRouteName: 'LeftHandNav',
   screens: {
-    Chat: 'r/:id?',
+    Chat: { 
+      path: 'r/:id?', 
+      parse: { id: (id) => parseInt(id) } 
+    },
     LeftHandNav: '',
     SettingsStack: {
+      path: 'settings',
       screens: {
-        Settings: 'settings',
-        About: 'settings/about',
+        Settings: '',
+        About: 'about',
       },
     },
     Search: 'search'
   },
 };
 
-const linking = {
-  prefixes: ['http://localhost', 'https://navigation-test-lime.vercel.app'],
-  config,
-  getStateFromPath(path, cfg) {
-    const state = getStateFromPath(path, cfg);
-    console.log('Get state from path: ', state);
 
-    // We need to add the LHN if it does not exist
-    if (!_.find(state.routes, r => r.name === 'LeftHandNav')) {
-      state.routes.splice(0, 0, {name: 'LeftHandNav'});
+const fixState = (state) => {
+    // we don't want to add Chat route for small screens
+    if (checkIsSmallScreen(Dimensions.get('window').width)) {
+      return 
     }
-
-    if (!_.find(state.routes, r => r.name === 'Chat')) {
+    if (!find(state.routes, r => r.name === 'Chat')) {
       state.routes.splice(1, 0, {name: 'Chat', params: {id: 1}});
     }
+}
 
+const linking = {
+  prefixes: ['http://localhost', 'https://navigation-test-lime.vercel.app', Linking.createURL('/')],
+  config,
+  getStateFromPath(path, cfg) {
+    const state = getStateFromPath(path, cfg);
+    fixState(state)
     return state;
   }
 };
 
-const isSmallScreenWidth = Dimensions.get('window').width <= 800;
-
-/**
- * By default the back handler will pop. We want it to go "back" instead of "up". All screens that are inside a nested navigator
- * except the root screen of that nested navigator will need to implement this.
- */
-function WithCustomBackBehavior(props) {
-  useFocusEffect(() => {
-    const subscription = BackHandler.addEventListener('hardwareBackPress', () => {
-      props.navigation.popToTop();
-    });
-
-    return () => subscription.remove();
-  });
```

Comment:
It looks like we don't need this anymore

Reply:
Yeah, the app seems to work properly for Android.